packages/keystone/cipher.js

```diff
@@ -0,0 +1,157 @@
+const crypto = require('crypto')
+
+const { isEmpty } = require('lodash')
+
+// since data is converted in hex, ':' shouldn't be in it
+const SEPARATOR = ':'
+const SUPPORTED_MODES = ['cbc', 'ctr', 'cfb', 'ofb', 'ecb']
+const SUGGESTIONS = {
+    'ecb': 'WARNING! Don\'t use "ecb" in real application as it is insecure!',
+    'cfb': 'Please, consider using "ctr" or "cbc"',
+    'ofb': 'Please, consider using "ctr" or "cbc"',
+}
+
+class CipherManager {
+
+    /** @type {{[version: string]: {cipher: string, secret: string, ivLength: number}}} */
+    versions = {}
+    currentVersionKey
+    
+    /**
+     * @param {Array<{version: string, cipher: string, secret: string}>} options
+     * @param {string?} currentVersionKey
+     */
+    constructor (options, currentVersionKey) {
+        if (!options) {
+            throw new Error('versions is required')
+        }
+        if (!Array.isArray(options) || options.length === 0) {
+            throw new Error('versions must be not empty array')
+        }
+
+        this._validateVersions(options)
+
+        if (currentVersionKey === null || currentVersionKey === undefined) {
+            throw new Error('currentVersionKey is required')
+        }
+        if (!this.versions[currentVersionKey]) {
+            throw new Error('invalid currentVersionKey: can\'t find version with this key')
+        }
+        this.currentVersionKey = currentVersionKey
+    }
+
+    /**
+     * @typedef CipherManagerResult
+     * @property {string} version
+     * @property {CipherManagerResultIv} iv
+     */
+
+    /**
+     * @typedef CipherManagerResultIv
+     * @property {string} value
+     * @property {BufferEncoding} encoding
+     */
+
+    /**
+     * If config.versionKey is not provided, currentVersionKey being used
+     * @param {string} data
+     * @param {Partial<CipherManagerResult>?} config
+     * @returns {CipherManagerResult & {encrypted:string}}
+     */
+    encrypt (data, config = {}) {
```

Comment:
Why the config is not a part of constructor

Reply:
Needed for overrides, but i simplified and removed it